recipes/gm2calc/all/conanfile.py

```diff
@@ -1,74 +1,94 @@
 import os
-import conan
-from conans import ConanFile, CMake, tools
 
-required_conan_version = ">=1.30.0"
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir, replace_in_file, save
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.53.0"
 
 
 class Gm2calcConan(ConanFile):
     name = "gm2calc"
+    description = "C++ library to calculate the anomalous magnetic moment of the muon in the MSSM and 2HDM"
     license = "GPL-3.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/GM2Calc/GM2Calc"
-    description = "C++ library to calculate the anomalous magnetic moment of the muon in the MSSM and 2HDM"
     topics = ("high-energy", "physics", "hep", "magnetic moment", "muon", "mssm", "2hdm")
-    settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
 
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
 
     def export_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.75.0")
-        self.requires("eigen/3.3.9")
+        self.requires("boost/1.83.0")
+        self.requires("eigen/3.4.0", transitive_headers=True)
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 11)
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        conan.tools.files.rename(self, "GM2Calc-{}".format(self.version), self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
```

Comment:
```suggestion
        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
        tc.variables["ENABLE_TESTS"] = True
```
Since 2.2 you can actually disable tests: 
https://github.com/GM2Calc/GM2Calc/blob/v2.2.0/CMakeLists.txt#L9